R/lint.R

```diff
@@ -390,6 +390,16 @@ validate_linter_object <- function(linter, name) {
   linter
 }
 
+is_linter_factory <- function(fun) {
+  # A linter factory is a function whose last call is to Linter()
+  bdexpr <- body(fun)
+  # covr internally transforms each call into if (TRUE) { covr::count(...); call }
```

Comment:
nit: maybe better to move that out of the `while()` condition since it only need to be tested once at most?

OTOH, evaluating this while loop more than once is already in funky territory -- means authors intentionally writing adversarial code AFAICT...

Reply:
I wanted to stick close to the S3 generic detection here, since that has been well tested.

Reply:
works for me :+1: